internal/db/data_flow.md

```diff
@@ -0,0 +1,544 @@
+# DefraDB Data Flow: Request Lifecycle
```

Comment:
todo: Mentioned by Fred in the standup, and I agree - this does need to be renamed.  I think (for an overview doc) the current scope is good, but the name does not reflect this.  Neither does the first paragraph.